src/lib/mdns/Resolver_ImplMinimalMdns.cpp

```diff
@@ -161,48 +271,82 @@ void PacketDataReporter::OnResource(ResourceType type, const ResourceData & data
     ///    - Can extract: fabricid, nodeid, port
     ///    - References ServerName
     /// - Additional records tied to ServerName contain A/AAAA records for IP address data
-
-    if (data.GetType() == QType::SRV)
+    switch (data.GetType())
     {
+    case QType::SRV: {
         SrvRecord srv;
-
         if (!srv.Parse(data.GetData(), mPacketRange))
         {
             ChipLogError(Discovery, "Packet data reporter failed to parse SRV record");
             mHasNodePort = false;
         }
-        else
+        else if (mDiscoveryType == DiscoveryType::kOperational)
+        {
+            OnOperationalSrvRecord(data.GetName(), srv);
+        }
+        else if (mDiscoveryType == DiscoveryType::kCommissionableNode)
         {
-            OnSrvRecord(data.GetName(), srv);
+            OnCommissionableNodeSrvRecord(data.GetName(), srv);
         }
+        break;
     }
-    else if (data.GetType() == QType::A)
-    {
-        chip::Inet::IPAddress addr;
-
+    case QType::TXT:
+        if (mDiscoveryType == DiscoveryType::kCommissionableNode)
+        {
+            TxtRecordDelegateImpl textRecordDelegate(&mCommissionableNodeData);
+            ParseTxtRecord(data.GetData(), &textRecordDelegate);
+        }
+        break;
+    case QType::A: {
+        Inet::IPAddress addr;
         if (!ParseARecord(data.GetData(), &addr))
         {
             ChipLogError(Discovery, "Packet data reporter failed to parse A record");
             mHasIP = false;
         }
         else
         {
-            OnIPAddress(addr);
+            if (mDiscoveryType == DiscoveryType::kOperational)
+            {
+                OnOperationalIPAddress(addr);
+            }
+            else if (mDiscoveryType == DiscoveryType::kCommissionableNode)
+            {
+                OnCommissionableNodeIPAddress(addr);
+            }
         }
+        break;
     }
-    else if (data.GetType() == QType::AAAA)
-    {
-        chip::Inet::IPAddress addr;
-
+    case QType::AAAA: {
+        Inet::IPAddress addr;
         if (!ParseAAAARecord(data.GetData(), &addr))
         {
             ChipLogError(Discovery, "Packet data reporter failed to parse A record");
```

Comment:
```suggestion
            ChipLogError(Discovery, "Packet data reporter failed to parse AAAA record");
```

Reply:
changed when I was making changes in the next commit